spec/pagseguro/transaction_request/request_serializer_spec.rb

```diff
@@ -167,47 +487,100 @@ def build_item(index)
           district: "DISTRICT",
           number: "NUMBER",
           complement: "COMPLEMENT"
-        }
-      })
+        })
+
+        allow(transaction_request).to receive(:billing_address).and_return(address)
+      end
+
+      it { expect(params).to include(billingAddressStreet: "STREET") }
+      it { expect(params).to include(billingAddressCountry: "BRA") }
+      it { expect(params).to include(billingAddressState: "STATE") }
+      it { expect(params).to include(billingAddressCity: "CITY") }
+      it { expect(params).to include(billingAddressPostalCode: "POSTAL_CODE") }
+      it { expect(params).to include(billingAddressDistrict: "DISTRICT") }
+      it { expect(params).to include(billingAddressNumber: "NUMBER") }
+      it { expect(params).to include(billingAddressComplement: "COMPLEMENT") }
     end
 
-    it { expect(params).to include(shippingType: 1) }
-    it { expect(params).to include(shippingCost: "1234.56") }
-    it { expect(params).to include(shippingAddressStreet: "STREET") }
-    it { expect(params).to include(shippingAddressCountry: "BRA") }
-    it { expect(params).to include(shippingAddressState: "STATE") }
-    it { expect(params).to include(shippingAddressCity: "CITY") }
-    it { expect(params).to include(shippingAddressPostalCode: "POSTAL_CODE") }
-    it { expect(params).to include(shippingAddressDistrict: "DISTRICT") }
-    it { expect(params).to include(shippingAddressNumber: "NUMBER") }
-    it { expect(params).to include(shippingAddressComplement: "COMPLEMENT") }
-  end
+    context "sender serialization" do
+      before do
+        sender = PagSeguro::Sender.new({
+          hash: "HASH",
+          email: "EMAIL",
+          name: "NAME",
+          cpf: "CPF",
+          phone: {
+            area_code: "AREA_CODE",
+            number: "NUMBER"
+          }
+        })
 
-  context "installment serialization" do
-    before do
-      installment = PagSeguro::TransactionInstallment.new({
-        value: "459.50",
-        quantity: "1"
-      })
+        allow(transaction_request).to receive(:sender).and_return(sender)
+      end
 
-      allow(transaction_request).to receive(:installment).and_return(installment)
+      it { expect(params).to include(senderHash: "HASH") }
+      it { expect(params).to include(senderEmail: "EMAIL") }
+      it { expect(params).to include(senderName: "NAME") }
+      it { expect(params).to include(senderCPF: "CPF") }
+      it { expect(params).to include(senderAreaCode: "AREA_CODE") }
+      it { expect(params).to include(senderPhone: "NUMBER") }
     end
 
-    it { expect(params).to include(installmentValue: "459.50") }
-    it { expect(params).to include(installmentQuantity: "1") }
-  end
+    context "shipping serialization" do
+      before do
```

Comment:
Perhaps you can try to use before(:all) because your examples only verify if params include something so whether you use before, it will do every time for each case.